server/src/main/java/org/opensearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -88,7 +90,7 @@
     private final SearchTransportService searchTransportService;
     private final Executor executor;
     private final ActionListener<SearchResponse> listener;
-    private final SearchRequest request;
+    private final PipelinedRequest request;
```

Comment:
I am wondering if we should instead inherit `PipelinedRequest` from SearchRequest` and delegate all the method call to the transformed search request held by the `PipelinedRequest`. That will avoid the signature change everywhere.

Reply:
@msfroh your thoughts?

Reply:
I'm not a big fan, since the methods of `SearchRequest` may be subject to change as we develop new search features. We'd then have to remember to make sure `PipelinedRequest` delegates any new methods. That said, it looks like `SearchRequest` hasn't changed much over the past couple of years.

Since `SearchRequest` has fields, subclassing + delegation isn't a great pattern -- we'd end up with two objects with a bunch of fields, but one delegates to the other. A better pattern would probably involve subclassing `SearchRequest` and using the shallow "copy constructor": https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/search/SearchRequest.java#L129

Reply:
@reta  -- The `PipelinedRequest` wrapper was your suggestion. Do you have an opinion on subclassing `SearchRequest` versus passing `PipelinedRequest` (and unwrapping the underlying `SearchRequest`) wherever needed?

My inclination is still to take the approach from this PR, where we keep the wrapper, and take the hit now on updating the `AbstractSearchAsyncAction` subclasses to take a `PipelinedRequest`. It looks bad in this PR, since it's a bunch of classes changing, but I think it's likely to be a one-time hit. Those words could come back to bite me in the future, though.

Reply:
@msfroh I had the same thought with subclassing but wanted to check if there is any other way we can change it. With `PipelineRequest` now any usage of the `SearchRequest` field will change to `pipelineRequest.getTransformedRequest().getField()`. Whereas with delegation it can potentially remain as is but I think it may be error prone if `PipelineRequest` inherits `SearchRequest` and forgets to override any method of a `SearchRequest`. Current approach seems better that way.

Reply:
@msfroh sorry, haven't see your comment

> The PipelinedRequest wrapper was your suggestion. Do you have an opinion on subclassing SearchRequest versus passing PipelinedRequest (and unwrapping the underlying SearchRequest) wherever needed?

The usage `PipelinedRequest` becomes quite invasive for a reason to call `tansform()` all the time, subclassing `SearchRequest` to  in this case would make be cleaner.

Reply:
Sounds good. We can make `PipelinedSearchRequest` a subclass of `SearchRequest` and maybe add `instanceof` checks before casting for the transform calls.

@navneet1v, do you want to make that change or would you like me to take it?

Reply:
@msfroh this change has more than naming change. You are right person to make the change.

So could you please do that.